components/eamxx/src/physics/iop_forcing/eamxx_iop_forcing_process_interface.cpp

```diff
@@ -0,0 +1,546 @@
+#include "physics/iop_forcing/eamxx_iop_forcing_process_interface.hpp"
+
+#include "share/property_checks/field_within_interval_check.hpp"
+
+namespace scream
+{
+// =========================================================================================
+void IOPForcing::set_grids(const std::shared_ptr<const GridsManager> grids_manager)
+{
+  using namespace ekat::units;
+
+  m_grid = grids_manager->get_grid("Physics");
+  const auto& grid_name = m_grid->name();
+
+  m_num_cols = m_grid->get_num_local_dofs(); // Number of columns on this rank
+  m_num_levs = m_grid->get_num_vertical_levels();  // Number of levels per column
+
+  // Define the different field layouts that will be used for this process
+  FieldLayout scalar2d     = m_grid->get_2d_scalar_layout();
+  FieldLayout scalar3d_mid = m_grid->get_3d_scalar_layout(true);
+  FieldLayout vector3d_mid = m_grid->get_3d_vector_layout(true,2);
+
+  constexpr int pack_size = Pack::n;
+
+  add_field<Required>("ps", scalar2d, Pa, grid_name);
+
+  add_field<Updated>("horiz_winds", vector3d_mid, m/s, grid_name, pack_size);
+  add_field<Updated>("T_mid", scalar3d_mid, K, grid_name, pack_size);
+
+  add_tracer<Updated>("qv", m_grid, kg/kg, pack_size);
+  add_group<Updated>("tracers", grid_name, pack_size, Bundling::Required);
+
+  // Sanity check that iop data manager is setup by driver
+  EKAT_REQUIRE_MSG(m_iop_data_manager,
+    "Error! IOPDataManager not setup by driver, but IOPForcing"
+    "being used as an ATM process.\n");
+
+  // Compute number of buffer views
+  const auto iop_nudge_tq = m_iop_data_manager->get_params().get<bool>("iop_nudge_tq");
+  const auto iop_nudge_uv = m_iop_data_manager->get_params().get<bool>("iop_nudge_uv");
+  if (iop_nudge_tq) m_buffer.num_1d_scalar_nlev += 2;
+  if (iop_nudge_uv) m_buffer.num_1d_scalar_nlev += 2;
+}
+// =========================================================================================
+void IOPForcing::
+set_computed_group_impl (const FieldGroup& group)
+{
+  EKAT_REQUIRE_MSG(group.m_info->size() >= 1,
+                   "Error! IOPForcing requires at least qv as tracer input.\n");
+
+  const auto& name = group.m_info->m_group_name;
+
+  EKAT_REQUIRE_MSG(name=="tracers",
```

Comment:
FYI, before calling the impl method here, the base class already checks (in `set_computed_group`) that the group name is among the list of groups that was requested.

Reply:
I think this function is probably not necessary for this process since it only serves to set `m_num_tracers` which can be done in `initialize_impl()`.